source/network/network.go

```diff
@@ -110,26 +119,28 @@ func (s *networkSource) GetFeatures() *nfdv1alpha1.Features {
 	return s.features
 }
 
-func detectNetDevices() ([]nfdv1alpha1.InstanceFeature, error) {
+func detectNetDevices() ([]nfdv1alpha1.InstanceFeature, []nfdv1alpha1.InstanceFeature, error) {
 	sysfsBasePath := hostpath.SysfsDir.Path(sysfsBaseDir)
 
 	ifaces, err := os.ReadDir(sysfsBasePath)
 	if err != nil {
-		return nil, fmt.Errorf("failed to list network interfaces: %w", err)
+		return nil, nil, fmt.Errorf("failed to list network interfaces: %w", err)
 	}
 
 	// Iterate over devices
-	info := make([]nfdv1alpha1.InstanceFeature, 0, len(ifaces))
+	devIfacesinfo := make([]nfdv1alpha1.InstanceFeature, 0, len(ifaces))
+	virtualIfacesinfo := make([]nfdv1alpha1.InstanceFeature, 0, len(ifaces))
+
 	for _, iface := range ifaces {
 		name := iface.Name()
 		if _, err := os.Stat(filepath.Join(sysfsBasePath, name, "device")); err == nil {
-			info = append(info, readIfaceInfo(filepath.Join(sysfsBasePath, name), ifaceAttrs))
+			devIfacesinfo = append(devIfacesinfo, readIfaceInfo(filepath.Join(sysfsBasePath, name), devIfaceAttrs))
 		} else if klogV := klog.V(3); klogV.Enabled() {
-			klogV.InfoS("skipping non-device iface", "interfaceName", name)
+			virtualIfacesinfo = append(virtualIfacesinfo, readIfaceInfo(filepath.Join(sysfsBasePath, name), virtualIfaceAttrs))
```

Comment:
We should not depend on the log verbosity on whether to scan virtual ifaces 😅

Reply:
ow 😣 , fixed!